modules/home-manager/neovim.nix

```
@@ -1,16 +1,22 @@
 { catppuccinLib }:
-{
-  config,
-  lib,
-  ...
-}:
-
+{ config, lib, ... }:
 let
   cfg = config.catppuccin.nvim;
-in
+  pluginSettings = lib.generators.toLua { } (defaultConfig // cfg.extraConfig);
 
+  defaultConfig = {
+    compile_path = lib.generators.mkLuaInline "compile_path";
+    flavour = cfg.flavor;
+  };
+in
 {
-  options.catppuccin.nvim = catppuccinLib.mkCatppuccinOption { name = "neovim"; };
+  options.catppuccin.nvim = catppuccinLib.mkCatppuccinOption { name = "neovim"; } // {
+    extraConfig = lib.mkOption {
```

```suggestion
    settings = lib.mkOption {
```

`extraConfig` usually implies a strongly typed set of options. What we're doing here is more akin to [RFC 42](https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md), which encourages the use of `settings` for freeform options